File: 프로그래머스/2/140107. 점 찍기/점 찍기.js
function solution(k, d) {
    let answer = 0;

    for (let x = 0; x <= d; x += k) {
        // 가능한 y의 최대 거리 계산
        const maxY = Math.floor(Math.sqrt(d * d - x * x));
        // y는 0, k, 2k, ... maxY 이하이므로 개수는 (maxY / k) + 1
        answer += Math.floor(maxY / k) + 1;
    }

    return answer;
}

File: 프로그래머스/2/140107. 점 찍기/점 찍기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <math.h> // sqrt 함수 사용

long long solution(int k, int d) {
    long long answer = 0;
    for (long long x = 0; x <= d; x += k) {
        long long max_y = (long long)sqrt((long long)d * d - x * x);
        answer += (max_y / k) + 1; // 0 포함이므로 +1
    }
    return answer;
}
